pkg/processor/openfaas.go

```diff
@@ -0,0 +1,117 @@
+package processor
+
+import (
+	"encoding/json"
+	"log"
+	"time"
+
+	sdk "github.com/openfaas-incubator/connector-sdk/types"
+	"github.com/openfaas/faas-provider/auth"
+	"github.com/pkg/errors"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/connection"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/events"
+
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	topicDelimiter     = ","
+	rebuildInterval    = time.Second * 10
+	timeout            = time.Second * 15
+	authMethodOpenFaaS = "basic_auth" // only this method is supported by the processor
+)
+
+// openFaaS implements the Processor interface
+type openFaaS struct {
```

Comment:
capital `OpenFaaS` for the struct can release `openFaaS` variable name

Reply:
I don't think we need to rename a struct so that we can use the name for a variable. I found that a struct and variable can have the same name. https://play.golang.org/p/KDF1_kssILZ

Although, I do like the idea of lowercasing the entire name, so it's `openfaas`.

Reply:
Went with `openfaasProcessor` and `func (openfaas *openfaasProcessor) ...`